### PR TITLE
Hassan/characterize events

### PR DESCRIPTION
**Pull request by HassanJaveed84**
@mulby @brianhw @tobz 

These are the tasks that I ran as part of my story.

### CODE REVIEW

File: edx/analytics/tasks/monitor/characterize_events.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    bucket = luigi.Parameter()
+
+    key_prefix = luigi.Parameter(default='logs/tracking/')
+
+
+    url_to_timestamp_map = {}
+
+    def requires_local(self):
+        return ListS3FilesWithDateTask(
```
Comment (brianhw) [inline edx/analytics/tasks/monitor/characterize_events.py:52]: Is there any reason to call super here?   I'm looking at events_obfuscate.py as a model of something that works.

Comment (brianhw) [reply edx/analytics/tasks/monitor/characterize_events.py:52]: And then you could add the target explicitly to the requirements as a dict entry.  If super doesn't return a dict, then create an empty one and add the super's return to that new dict.  Then add this requirement as a particular key value in the returned dict.

File: edx/analytics/tasks/monitor/characterize_events.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            key_prefix=self.key_prefix,
+            output_root=self.output_root,
+        )
+
+    def init_local(self):
+        super(CharacterizeEventsTask, self).init_local()
+        with self.input_local().open() as f_in:
+            lines = f_in.readlines()
+            for line in lines:
+                file_url, timestamp = line.split()
```
Comment (brianhw) [inline edx/analytics/tasks/monitor/characterize_events.py:64]: Does it make sense to split on \t instead?  That wouldn't explain your issue, but...

File: edx/analytics/tasks/monitor/characterize_events.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            bucket=self.bucket,
+            key_prefix=self.key_prefix,
+            output_root=self.output_root,
+        )
+
+    def init_local(self):
+        super(CharacterizeEventsTask, self).init_local()
+        with self.input_local().open() as f_in:
+            lines = f_in.readlines()
+            for line in lines:
```
Comment (brianhw) [inline edx/analytics/tasks/monitor/characterize_events.py]: nit: you could do 'for line in f_in',  I believe.

File: edx/analytics/tasks/monitor/characterize_events.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                file_url, timestamp = line.split()
+                self.url_to_timestamp_map[file_url] = timestamp
+
+    def mapper(self, line):
+        input_file = os.environ['mapreduce_map_input_file']
```
Comment (brianhw) [inline edx/analytics/tasks/monitor/characterize_events.py]: nit:  use self.get_map_input_file()

File: edx/analytics/tasks/monitor/characterize_events.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+    def requires_local(self):
+        return ListS3FilesWithDateTask(
+            bucket=self.bucket,
+            key_prefix=self.key_prefix,
+            output_root=self.output_root,
+        )
+
+    def init_local(self):
+        super(CharacterizeEventsTask, self).init_local()
```
Comment (brianhw) [inline edx/analytics/tasks/monitor/characterize_events.py]: I would probably try adding log statements in here as well, to confirm that it's being read.  That it's being called, and that the dict is getting populated.

**Review by brianhw — COMMENTED** [review]
I don't see anything wrong here that would explain why you are getting zero entries.  Again, it looks close enough to the filtering done in events_obfuscation, which (hopefully) works.

File: edx/analytics/tasks/monitor/characterize_events.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+class CharacterizeEventsTask(OverwriteOutputMixin, EventLogSelectionMixin, MapReduceJobTask):
+
+    output_root = luigi.Parameter()
+
+    bucket = luigi.Parameter()
+
+    key_prefix = luigi.Parameter(default='logs/tracking/')
+
+
+    url_to_timestamp_map = {}
```
Comment (HassanJaveed84) [inline edx/analytics/tasks/monitor/characterize_events.py]: We need to set the data structure on the object for it to be available on hadoop nodes.

**Pull request closed without merging**